xls/contrib/integrator/ir_integrator.cc

```diff
@@ -18,6 +18,122 @@
 
 namespace xls {
 
+absl::StatusOr<std::unique_ptr<IntegrationFunction>>
+IntegrationFunction::MakeIntegrationFunctionWithParamTuples(
+    Package* package, absl::Span<const Function* const> source_functions) {
+  // Create integration function object.
+  std::unique_ptr<IntegrationFunction> integration_function =
+      absl::make_unique<IntegrationFunction>();
+  integration_function->package_ = package;
+
+  // Create ir function.
+  // TODO(jbaileyhandle): Make function name an optional argument.
+  static int64 integration_functions_count = 0;
```

Comment:
It'd be better not to have a static variable here as the result of the method is no longer just a function of its inputs. Fine with the TODO to fix later.

Reply:
Made the name a function argument.